pages/articles/SectionsHandler.inc.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/**
+ * @file pages/sections/SectionsHandler.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SectionsHandler
+ * @ingroup pages_articles
+ *
+ * @brief Handle requests for sections functions.
+ *
+ */
+
+use APP\handler\Handler;
+
+use APP\template\TemplateManager;
+use PKP\submission\PKPSubmission;
+
+class SectionsHandler extends Handler
+{
+    /** sections associated with the request **/
+    public $sections;
+
+    /**
+     * @copydoc PKPHandler::authorize()
+     */
+    public function authorize($request, &$args, $roleAssignments)
+    {
+        import('lib.pkp.classes.security.authorization.ContextRequiredPolicy');
+        $this->addPolicy(new ContextRequiredPolicy($request));
+
+        import('classes.security.authorization.OjsJournalMustPublishPolicy');
+        $this->addPolicy(new OjsJournalMustPublishPolicy($request));
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    /**
+     * View a section
+     *
+     * @param $args array [
+     *		@option string Section ID
+     *		@option string page number
+     * ]
+     *
+     * @param $request PKPRequest
+     *
+     * @return null|JSONMessage
+     */
+    public function section($args, $request)
+    {
+        $sectionPath = $args[0] ?? null;
+        $page = isset($args[1]) && ctype_digit((string) $args[1]) ? (int) $args[1] : 1;
```

Comment:
I think that should be the case, of course the migration code is now missing for generating the url paths for old sections

Reply:
If that's the case, then remove the check for a URL containing an `id`. We don't need it.

Reply:
Do you mean the ctype_digit checks? Those do not apply to the `urlPath`, they are related to the page number of the archive view.

Reply:
@NateWr while writing the code for creating paths during migration I found this https://github.com/cocur/slugify. What do you think?

Secondly, I noticed that both OMP and OPS call this `path` instead of `urlPath`.

Reply:
I use stringy in the conversion now. Since this is just done once during the upload and the result can be edited later, I do not think it is that critical.